File: frontend/src/api/admin/AdminCommentsApi.ts
import { DeleteCommentRequestParamsSchema, DeleteCommentResponseDataSchema, GetAdminCommentsRequestQuerySchema, GetAdminCommentsResponseDataSchema, type AdminComment } from "../api-contracts.js";
import { request } from "../client.js";

export const getAdminComments = async (userId: string): Promise<AdminComment[]> => {
  GetAdminCommentsRequestQuerySchema.parse({});
  return request("/admin/comments", { method: "GET", headers: { "x-user-id": userId } }, GetAdminCommentsResponseDataSchema);
};

export const deleteComment = async (userId: string, commentId: string): Promise<AdminComment> =>
  request(`/admin/comments/${DeleteCommentRequestParamsSchema.parse({ commentId }).commentId}`, { method: "DELETE", headers: { "x-user-id": userId } }, DeleteCommentResponseDataSchema);

File: backend/microservice/src/admin/api/AdminCommentsApi.scala
package microservice.admin.api

object GetAdminCommentsEndpoint {
  val name: String = "GetAdminComments"
  val method: String = "GET"
  val path: String = "/admin/comments"
  val businessLogic: String =
    "管理员读取全量评论，按创建时间倒序排列。"
}

object DeleteCommentEndpoint {
  val name: String = "DeleteComment"
  val method: String = "DELETE"
  val path: String = "/admin/comments/:commentId"
  val businessLogic: String =
    "管理员按 commentId 删除评论并返回被删除的评论对象。"
}
